src/persist-client/src/cfg.rs

```diff
@@ -247,26 +352,65 @@ impl ConsensusKnobs for PersistConfig {
 pub struct PersistParameters {
-    /// Configures [`PersistConfig::blob_target_size`].
+    /// Configures [`DynamicConfig::blob_target_size`].
     pub blob_target_size: Option<usize>,
-    /// Configures [`PersistConfig::compaction_minimum_timeout`].
+    /// Configures [`DynamicConfig::compaction_minimum_timeout`].
     pub compaction_minimum_timeout: Option<Duration>,
 }
 
 impl PersistParameters {
     /// Update the parameter values with the set ones from `other`.
     pub fn update(&mut self, other: PersistParameters) {
-        if let Some(v) = other.blob_target_size {
-            self.blob_target_size = Some(v);
+        // Deconstruct self and other so we get a compile failure if new fields
+        // are added.
+        let Self {
+            blob_target_size: self_blob_target_size,
+            compaction_minimum_timeout: self_compaction_minimum_timeout,
+        } = self;
+        let Self {
+            blob_target_size: other_blob_target_size,
+            compaction_minimum_timeout: other_compaction_minimum_timeout,
+        } = other;
+        if let Some(v) = other_blob_target_size {
+            *self_blob_target_size = Some(v);
         }
-        if let Some(v) = other.compaction_minimum_timeout {
-            self.compaction_minimum_timeout = Some(v);
+        if let Some(v) = other_compaction_minimum_timeout {
+            *self_compaction_minimum_timeout = Some(v);
         }
     }
 
     /// Return whether all parameters are unset.
     pub fn all_unset(&self) -> bool {
-        self.blob_target_size.is_none() && self.compaction_minimum_timeout.is_none()
+        // TODO: Where is this called? We could save a tiny bit of boilerplate
+        // by comparing self to Self::default().
+        //
+        // Deconstruct self so we get a compile failure if new fields are added.
+        let Self {
+            blob_target_size,
+            compaction_minimum_timeout,
+        } = self;
+        blob_target_size.is_none() && compaction_minimum_timeout.is_none()
+    }
+
+    /// Applies the parameter values to persist's in-memory config object.
+    ///
+    /// Note that these overrides are not all applied atomically: i.e. it's
+    /// possible for persist to race with this and see some but not all of the
+    /// parameters applied.
+    pub fn apply(&self, cfg: &PersistConfig) {
+        // Deconstruct self so we get a compile failure if new fields are added.
+        let Self {
+            blob_target_size,
+            compaction_minimum_timeout,
+        } = self;
+        if let Some(blob_target_size) = blob_target_size {
+            cfg.dynamic
+                .blob_target_size
+                .store(*blob_target_size, DynamicConfig::STORE_ORDERING);
+        }
+        if let Some(_compaction_minimum_timeout) = compaction_minimum_timeout {
+            // TODO: Figure out how to represent Durations in DynamicConfig.
+        }
```

Comment:
If it's hard to apply `compaction_minimum_timeout` dynamically, how about removing it as a `SystemVar` and `PersistParameter` for now? It served its purpose as a test case for having duration parameters, I don't think we need it for that anymore. On the other hand, it is confusing to have a flag in Launch Darkly that doesn't have an effect when changed.

Reply:
Ah, I wasn't clear on how big of a lift removing this TODO was. I'm fine with leaving the `SystemVar` unimplemented for a bit longer!